api/src/Page/Contact.php

```diff
@@ -46,9 +46,24 @@ function _get_contacts() {
                 $where .= ' AND c.labcontactid=:'.(sizeof($args)+1);
                 array_push($args, $this->arg('cid'));
             }
-            
 
-            $tot = $this->db->pq("SELECT count(c.labcontactid) as tot FROM labcontact c  $where", $args);
+            if ($this->has_arg('s')) {
+                $st = sizeof($args) + 1;
```

Comment:
This pattern is repeated everywhere, lets not do it again. Instead we should pass `s` and a list of field names to a `whereSearch` method of `WhereClauseQueryBuilder` which is probably part of, or at least used by, `DatabaseQueryBuilder`. The query builder can be used directly here or even extended to do the whole paginate search.
Would love to see the pageing code but in there too.
This function should be something like:

```
query.where(proposalId, $this->proposalid)
if cid:
  query.where(proposalId, $this->proposalid)
query.search($fields, this->arg('s')

$where = query.where_clause()
```